src/app/OperationalDeviceProxy.cpp

```diff
@@ -354,7 +354,7 @@ void OperationalDeviceProxy::OnFirstMessageDeliveryFailed()
 
 void OperationalDeviceProxy::OnSessionHang()
 {
-    // TODO: establish a new session
+    Disconnect();
```

Comment:
I thought based on our discussions, that we were going to mark the session as defunct straight from within `ExchangeContext::NotifyResponseTimeout` and from `ReliableMessageMgr`, in addition to calling `OnSessionHang`.

With the changes that @tehampson is doing, `OperationalDeviceProxy` is pivoting to only being resident while a session is being established, and so this is effectively going to be a no-op pretty much 99% of the time.

Reply:
Importantly, the marking of a session as defunct should not only happen on the controller side - it should apply equally to sessions used by server objects (like `ReadHandler` for example too, when it generates a report, that should fail because the session is now defunct).

Reply:
We would need to do both, then, until @tehampson's changes happen, because just marking the session defunct does not help if we have an operational device proxy still hanging around that references it that we hand out to people.

I can certainly do that.